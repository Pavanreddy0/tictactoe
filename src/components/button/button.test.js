import { render, cleanup } from "@testing-library/react";
import Button from "./button";
import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";

afterEach(cleanup);

it("button renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<Button />, div);
});

it("button renders successfully", () => {
  const { getByTestId } = render(<Button value="Undo" />);
  expect(getByTestId("btn")).toHaveTextContent("Undo");
});

it("matches snapshow", () => {
  const tree = renderer.create(<Button value="Refresh"></Button>).toJSON();
  expect(tree).toMatchSnapshot();
});

it("matches snapshot 2", () => {
  const tree = renderer.create(<Button value="Pavan"></Button>).toJSON();
  expect(tree).toMatchSnapshot();
});
