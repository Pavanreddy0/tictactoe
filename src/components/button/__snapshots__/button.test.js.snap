// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 2 1`] = `
<div>
  <button
    data-testid="btn"
    name="btn"
    style={
      Object {
        "cursor": "pointer",
        "display": "grid",
        "fontSize": "20px",
        "fontWeight": "80",
        "height": "30px",
        "margin": "10px auto",
        "width": "250px",
      }
    }
  >
    Pavan
  </button>
</div>
`;

exports[`matches snapshow 1`] = `
<div>
  <button
    data-testid="btn"
    name="btn"
    style={
      Object {
        "cursor": "pointer",
        "display": "grid",
        "fontSize": "20px",
        "fontWeight": "80",
        "height": "30px",
        "margin": "10px auto",
        "width": "250px",
      }
    }
  >
    Refresh
  </button>
</div>
`;
